src/components/heatmap-chart/HeatmapChart.tsx

```diff
@@ -0,0 +1,197 @@
+import * as React from 'react';
+
+export interface Title {
+    text: string;
+}
+
+export interface XAxis {
+    categories: string[];
+}
+
+export interface YAxis {
+    categories: string[];
+}
+
+export interface DataLabel {
+    enabled: boolean;
+    colorCode: string;
+}
+
+export interface Series {
+    name: string;
+    borderWidth: number;
+    data: number[][];
+    dataLabels: DataLabel;
+}
+
+export interface HeatmapChartData {
+    title: Title;
+    xAxis: XAxis;
+    yAxis: YAxis;
+    width: number;
+    height: number;
+    series: Series;
+    colorCode: string;
+}
+
+function Map(value: number, min1: number, max1: number, min2: number, max2: number) {
+    // (v - min1) / (max1 - min 1) = (result - min2) / (max2 - min2)
+    return min2 + (((max2 - min2) * (value - min1)) / (max1 - min1));
+}
+
+function Big(series: Series) {
+    let big = -Infinity;
+    for (const s of series.data) {
+        if (big < s[2]) {
+            big = s[2];
+        }
+    }
+    return big;
+}
+function Small(series: Series) {
+    let small = Infinity;
+    for (const s of series.data) {
+        if (small > s[2]) {
+            small = s[2];
+        }
+    }
+    return small;
+}
+
+function PlaceDataCells(xl: number, yl: number, w: number, h: number, s: Series) {
+    return ( s.data.map((i, index) => (
+        <>
+            <rect
+                key={index}
+                x={100 + i[0] * ((w - 250) / xl)}
+                y={50 + i[1] * ((h - 150) / yl)}
+                width={((w - 250) / xl)}
+                height={((h - 150) / yl)}
+                strokeWidth={s.borderWidth}
+                stroke="white"
+                fill={
+                    'rgb(' +
+                    (255 - Map(i[2], Small(s), Big(s), 0, 255)) +
+                    ',' +
+                    (255 - Map(i[2], Small(s), Big(s), 0, 255)) +
+                    ',255)'
+                }
+            />
+            {(s.dataLabels.enabled) ? (
+                <text
```

Comment:
We can use the common chart-label css class for <text> elements for consistency.

Reply:
I didn't get it

Reply:
you mean--- className="chart-label" ---

Reply:
Yup!